public/app/core/table_model.ts

```diff
@@ -1,17 +1,15 @@
 import _ from 'lodash';
+import { Column, TableData } from '@grafana/ui';
 
-interface Column {
-  text: string;
+// This class mutates and uses the extra column fields
+interface ColumnEX extends Column {
   title?: string;
-  type?: string;
   sort?: boolean;
   desc?: boolean;
-  filterable?: boolean;
-  unit?: string;
 }
 
-export default class TableModel {
-  columns: Column[];
+export default class TableModel implements TableData {
+  columns: ColumnEX[];
```

Comment:
EX - extended

Hopefully this is a private interface on a deprecated class.  These fields are only used from the angular TablePanel

Reply:
Can we call it `ColumnExtended` then? 🙂

Reply:
I changed it to `MutableColumn` -- these are the column properties that get changed in sort and by the angular table panel.  I hope we stop using these and manage them in state rather than the data model.